### PR TITLE
knative-serving definition of build images

### PR DESCRIPTION
**Pull request by mgencur**
FYI: The Makefile and dockerfiles are available in https://github.com/openshift/knative-serving/pull/5

I was not sure about two things:
1) if "test_binary_build_commands" is called before "images" so that I can reference "test-bin" image that would have the test binaries
2) if my Dockerfiles can sit in a pull request or have to be already merged to be picked by CI
Thanks for any feedback!

### CODE REVIEW

File: ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml [hunk 1]
```diff
@@ -12,6 +12,105 @@ build_root:
...
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/bin/webhook
+  to: activator
+- dockerfile_path: openshift/ci-operator/test/autoscale/Dockerfile
+  from: test-bin
+  inputs:
+    bin:
```
[the remark this answers is not in the document]
Comment (bbguimaraes) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml]: What are you trying to accomplish? These are all going to be promoted into the release image stream with this configuration.

File: ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml [hunk 2]
```diff
@@ -12,6 +12,105 @@ build_root:
...
+test_binary_build_commands: make test-install
+images:
+- dockerfile_path: openshift/ci-operator/knative-images/activator/Dockerfile
+  from: bin
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/bin/activator
+  to: activator
```
Comment (bbguimaraes) [inline ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml]: You have a job with `--promote`, so I assume these are supposed to be part of the release. If that's the case, we probably want to have more specific image names.

File: ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml [hunk 3]
```diff
@@ -12,6 +12,105 @@ build_root:
...
+- dockerfile_path: openshift/ci-operator/knative-images/autoscaler/Dockerfile
+  from: bin
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/bin/autoscaler
+  to: activator
```
[the remark this answers is not in the document]
Comment (mgencur) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml]: Oh. Will fix that.

File: ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml (hunk 2 above)
Comment (mgencur) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml]: My goal is currently to have all images available in some OpenShift Docker registry where I could pull them from when running my tests. For the `knative-test-images` we won't ever want them to be shipped to customers, they're only for testing. Regarding `knative-images` (which includes the poorly named activator) I'm currently not sure when we'll want to release them but sooner or later we'll want to do that.
Can you please give me a hint how to change the config for ci-operator to achieve that?

Comment (mgencur) [reply ci-operator/config/openshift/knative-serving/openshift-knative-serving-master.yaml]: OK. Small update after talking to my colleagues: we don't want to release the images. They're all for testing purposes.

**Comment by mgencur** [comment]
Thanks for suggestions. I think all concerns are addressed now after our conversation.

**Comment by bbguimaraes** [comment]
The configuration LGTM, I'll leave it to @smarterclayton to approve the promotion/imagestream part.

/lgtm
/hold
/assign @smarterclayton

**Comment by smarterclayton** [comment]
Promotion looks fine

**Comment by smarterclayton** [comment]
Actually, is there a reason you don’t promote to knative?

**Comment by mgencur** [comment]
@smarterclayton you mean why not this?
`name: knative`
We're currently setting CI for knative-serving, then we'll have knative-build and knative-eventing. Eventually, it might make sense to promote just to knative. We just need to see how many different image stream tags we'll have and discuss what makes sense.

**Comment by smarterclayton** [comment]
Generally an entire area has a single image stream.  OpenShift the entire
product has one image stream.  I would recommend doing the same for knative.

**Pull request merged**